react/ecmascript/02 - promises/index.js

```diff
@@ -1,9 +1,20 @@
-// Hint: use setInterval, create a new Promise and measure time with Date.now()
-
-export function delay() {
-
+export function delay(TIME) {
+  const start = Date.now();
+  const reason = new Error('This time is too much !');
+  return new Promise((resolve, reject) => {
+    if (TIME > 500) {
+      reject(reason);
+    } else {
+      setInterval(() => {
+        const delayedTime = Date.now() - start;
+        if (delayedTime <= TIME + 100 || delayedTime >= TIME - 100) {
```

Comment:
You don't need to check this. If `delayedTime` is within error bounds, the test will pass succesfully. If it goes out of time for some reason, this check won't fix that. You can remove the `if`.